lib/msf/core/payload/stager.rb

```diff
@@ -156,7 +156,14 @@ def handle_connection(conn, opts={})
     # If the stage should be sent over the client connection that is
     # established (which is the default), then go ahead and transmit it.
     if (stage_over_connection?)
-      p = generate_stage
+      uuid_raw = conn.get_once(16, 1)
+
+      opts = {}
+      if uuid_raw
+        opts[:uuid] = Msf::Payload::UUID.new({raw: uuid_raw})
```

Comment:
If a UUID is pulled from the socket, then we create an object from it and pass that through to everything else so that the UUID can be reused as part of the stages.